cli/cli.go

```diff
@@ -0,0 +1,466 @@
+
+	upstream := c.Bool("upstream")
+	downstream := c.Bool("downstream")
+
+	fields := parseFields(toxicFields)
+
+	p, err := t.Proxy(proxyName)
+	if err != nil {
+		fatalf("Failed to retrieve proxy %s: %s\n", proxyName, err.Error())
+	}
+
+	addToxic := func(stream string) {
+		t, err := p.AddToxic(toxicName, toxicType, stream, fields)
+		if err != nil {
+			fatalf("Failed to add toxic: %s\n", err.Error())
+		}
+		toxicName = t["name"].(string)
+		fmt.Printf("Added %s %s toxic '%s' on proxy '%s'\n", stream, toxicType, toxicName, proxyName)
+	}
+
+	if upstream {
+		addToxic("upstream")
+	}
+	// Default to downstream.
+	if downstream || (!downstream && !upstream) {
+		addToxic("downstream")
+	}
+}
+
+func updateToxic(c *cli.Context, t *toxiproxy.Client) {
+	proxyName := c.Args().First()
+	if proxyName == "" {
+		cli.ShowSubcommandHelp(c)
+		fatalf("Proxy name is required as the first argument.\n")
+	}
+	toxicName := getArgOrFail(c, "toxicName")
+	toxicFields := getArgOrFail(c, "fields")
+
+	fields := parseFields(toxicFields)
+
+	p, err := t.Proxy(proxyName)
+	if err != nil {
+		fatalf("Failed to retrieve proxy %s: %s\n", proxyName, err.Error())
+	}
+
+	_, err = p.UpdateToxic(toxicName, fields)
+	if err != nil {
+		fatalf("Failed to update toxic: %s\n", err.Error())
+	}
+
+	fmt.Printf("Updated toxic '%s' on proxy '%s'\n", toxicName, proxyName)
+}
+
+func removeToxic(c *cli.Context, t *toxiproxy.Client) {
+	proxyName := c.Args().First()
+	if proxyName == "" {
+		cli.ShowSubcommandHelp(c)
+		fatalf("Proxy name is required as the first argument.\n")
+	}
+	toxicName := getArgOrFail(c, "toxicName")
+
+	p, err := t.Proxy(proxyName)
+	if err != nil {
+		fatalf("Failed to retrieve proxy %s: %s\n", proxyName, err.Error())
+	}
+
+	err = p.RemoveToxic(toxicName)
+	if err != nil {
+		fatalf("Failed to remove toxic: %s\n", err.Error())
+	}
+
+	fmt.Printf("Removed toxic '%s' on proxy '%s'\n", toxicName, proxyName)
+}
+
+func parseFields(raw string) toxiproxy.Toxic {
+	parsed := map[string]interface{}{}
+	keyValues := strings.Split(raw, ",")
+	for _, keyValue := range keyValues {
+		kv := strings.SplitN(keyValue, "=", 2)
+		if len(kv) != 2 {
+			fatalf("Fields should be in format key1=value1,key2=value2\n")
+		}
+		value, err := strconv.Atoi(kv[1])
+		if err != nil {
+			fatalf("Toxic field was expected to be an integer.\n")
+		}
+		parsed[kv[0]] = value
+	}
+	return parsed
+}
+
+func enabledColor(enabled bool) string {
+	if enabled {
+		return greenColor
+	}
+
+	return redColor
+}
+
+func enabledText(enabled bool) string {
+	if enabled {
+		return "enabled"
+	}
+
+	return "disabled"
+}
+
+func listToxics(toxics toxiproxy.Toxics, stream string) {
+	fmt.Printf("%s%s toxics:\n%s", greenColor, stream, noColor)
+	if len(toxics) == 0 {
+		fmt.Printf("%sProxy has no %s toxics enabled.\n%s", redColor, stream, noColor)
+		return
+	}
+	for name, toxic := range toxics {
+		fmt.Printf("%s%s:%s", greenColor, name, noColor)
+		for property, value := range toxic {
```

Comment:
I think the field order will still be random, even with the changes in #92 
It would be nice to have a fixed order these.